### ISSUE TITLE
Themes not visible anymore in Safari firefox and iOS

### ISSUE THREAD

**Issue by Aham330** [basnijholt/lovelace-ios-themes#80]
after updating to 3.0.0 
all themes are not visible anymore. 
Tried refreshing pages in safari and firefox to no avail.

HA running on the latest version.

**Comment by dc911x** [comment]
I can confirm, on firefox no background image.

**Comment by basnijholt** [comment]
Did you try to restart Home Assistant?

**Comment by ekalle-swe** [comment]
Same as last update.
Only black on black background.
Yes, tried to restart ha and clear app cache.
I use dark green alt but that is only back/w after update

**Comment by basnijholt** [comment]
On the same network as your Home Assistant, can you view https://raw.githubusercontent.com/basnijholt/lovelace-ios-themes/a37376d918fcfe4785be99910dc9a7200ac37da9/themes/homekit-bg-blue-red.jpg without problems?

**Comment by Lumiaja** [comment]
Themes not working after update.

**Comment by Aham330** [comment]
> On the same network as your Home Assistant, can you view https://raw.githubusercontent.com/basnijholt/lovelace-ios-themes/a37376d918fcfe4785be99910dc9a7200ac37da9/themes/homekit-bg-blue-red.jpg without problems?

I'll try when back home next week.

**Comment by dc911x** [comment]
Still not working:

- Restarted HA
- Reinstalled v3 after downgrading again
- Removed cookies
- I can view (from the same network) https://raw.githubusercontent.com/basnijholt/lovelace-ios-themes/a37376d918fcfe4785be99910dc9a7200ac37da9/themes/homekit-bg-blue-red.jpg

Anything else I can try?

Reverting to previous version and a refresh is enough to make it work again.

**Comment by ekalle-swe** [comment]
> On the same network as your Home Assistant, can you view https://raw.githubusercontent.com/basnijholt/lovelace-ios-themes/a37376d918fcfe4785be99910dc9a7200ac37da9/themes/homekit-bg-blue-red.jpg without problems?

Yepp, no problem at all
![Screenshot_20240908_161858_com android chrome](https://github.com/user-attachments/assets/b73ecd4a-97d8-49b1-97be-176effc2d301)

**Comment by basnijholt** [comment]
Are you using the `-alternative` theme perhaps?

Sorry about these problems!

**Comment by basnijholt** [comment]
@Aham330 
> Sidenote: Is there a specific reason to keep the backgrounds somewhere else then locally.?

Yes, because now HACS doesn't download the files anymore, so users would have to manually download them and put them in `www/ios-themes/`

**Comment by basnijholt** [comment]
I opened a PR (#81) with a potential solution.

Could anyone try whether using the content of https://github.com/basnijholt/lovelace-ios-themes/blob/2d8700533e24661574cf517d8e5e43936b0a3b86/themes/ios-themes.yaml works for them?

Basically, just open the `/config/themes/ios-themes.yaml` file and paste the content of the link above.

If so, I will merge it and release 3.0.1.

**Comment by basnijholt** [comment]
I decided to merge it already because using a CDN would make it faster regardless.

Please let me know whether 3.0.1 works!

**Comment by ghost** [comment]
2409.1 brought with it some problem with aiohttp.
It has effected some integrations to reach their target. Shelly users has experienced problems reaching their devices. All my devices looked online but had the same state as before 9.1 upgrade. If you turned a switch to on, it would revet back to off in a few seconds and no action at all was detected in the Shelly app. If you turn something on or off in the app, ha would never hear it.
I have mine working again after a reboot of haos but some users say that it can come back again. As you are using http to reach the background on the fly, it could be the same problem.

https://github.com/home-assistant/core/issues/125465#issuecomment-2336881266

**Comment by dc911x** [comment]
@basnijholt After downloading 3.0.1 It still didn't work then I saw https://github.com/basnijholt/lovelace-ios-themes/issues/80#issuecomment-2336868976 after setting this to off and refreshing changing background did work.

Sorry about this, I only had "select theme" & "Dark mode" on my UI, seems in time the "Theme alternative" didn't make my dashboard anymore.

**Comment by Toggz** [comment]
Hi folks, 
I was seeing the same thing on 3.0.0 also but seems to be related to local pull of the backgrounds only (not the remote download of).
Try removing web cache of the HA app/browsers, that improved things for me on 3.0.0, after I was able to see the downloaded backgrounds from /local/. :)

**Comment by Aham330** [comment]
Hi All, 

Mean while I am home.

The link to the raw.images works fine for me.

I also removed my browser cache which does help for single pages. Meaning I would have to adjust all my 50 or so pages from blue-red-alternative to blue-red. Not looking forward to that exercise I decided trying to make blue-red default in home assistant and that did not work. Results in a black screen. 

@basnijholt I am not very technical in this respect but would it not be a lot easier and safer to have the backgrounds local. Even if it means that people have to copy some files into a folder location in HA?

When reverting back to 2.1.4 all works fine.

**Comment by hyxl520** [comment]
我也出现了类似问题，但是还没有解决

**Comment by hyxl520** [comment]
在回退到2.14后，恢复了正常

**Comment by elchininet** [comment]
Weird, it was working for me using the Github URLs (it is also working now using `jsdelivr`). Could this be related to WebView cache (independent from the App cache), [check this](https://github.com/elchininet/custom-sidebar/issues/130#issuecomment-2327025323) for more context.

**Comment by Toggz** [comment]
> Weird, it was working for me using the Github URLs (it is also working now using `jsdelivr`). Could this be related to WebView cache (independent from the App cache), [check this](https://github.com/elchininet/custom-sidebar/issues/130#issuecomment-2327025323) for more context.

Could well be. 
I had issues with HA version 2024.9.1 and reverted back to 2025.8.3 mainly to do with TTS.
With this custom integration also, after reverting back and refreshing cache on all devices, things went back to normal.

**Comment by Dreffi88** [comment]
Some input from me:
I am using some tablets with fully kiosk browser as wall panels for home assistant.
The background image couldn't be loaded on these devices. iPhone (HA App), iMac (Safari) etc. were working fine after update to 3.0.0 or 3.0.1.

Problem in my case: the tablets do not have internet access, because they usually don't need it. After I granted them access to the internet the background immediately appeared.

Maybe a special situation in my case, but you never know.

**Comment by ghost** [comment]
> Some input from me: I am using some tablets with fully kiosk browser as wall panels for home assistant. The background image couldn't be loaded on these devices. iPhone (HA App), iMac (Safari) etc. were working fine after update to 3.0.0 or 3.0.1.
> 
> Problem in my case: the tablets do not have internet access, because they usually don't need it. After I granted them access to the internet the background immediately appeared.
> 
> Maybe a special situation in my case, but you never know.

That's a relevant thought.
A wall panel, for dashboard only, should not be forced to have internet access.
Will not be maintained as often as a personal tablet/mobile/pc, so could be a security risk to let it have access.

**Comment by Dreffi88** [comment]
Additional info: the background remains available after it’s loaded once. I turned off internet access and rebooted the devices. The background is still there. 
if I change the theme, the background of the other theme is not available. When I turn back to the previous one, the background appears again. 

So it seems not to be a big problem for me. Maybe this info can help other users with an identical use case.

**Comment by vantech14** [comment]
Hi,

Latest 3.0.1 update with CDN still doesn't work for me. Rebooting home assistant and the devices fails to load the backgrounds.

Thanks.

**Comment by vandenbogerd** [comment]
Maybe some usefull info here: I also lack the background, during the day. 

When the phone switches to dark mode, the background appears.

**Comment by Dreffi88** [comment]
News: 
After some days without internet access the background is missing again.

**Comment by dc911x** [comment]
> @basnijholt After downloading 3.0.1 It still didn't work then I saw [#80 (comment)](https://github.com/basnijholt/lovelace-ios-themes/issues/80#issuecomment-2336868976) after setting this to off and refreshing changing background did work.
> 
> Sorry about this, I only had "select theme" & "Dark mode" on my UI, seems in time the "Theme alternative" didn't make my dashboard anymore.

Maybe my post got glanced over but I fixed it by disabling the "Theme alternative" to off!

![image](https://github.com/user-attachments/assets/feb065cc-bd71-42b0-9e38-a45e2d677e73)

**Comment by vantech14** [comment]
Upgraded again, same issues so will downgrade. Even tried to clear cache in the HA app

**Comment by elchininet** [comment]
>Upgraded again, same issues so will downgrade. Even tried to clear cache in the HA app

Did you try to clean the WebApp cache? Cleaning the App cache doesn't clean the WebApp cache.

Check [my comment](https://github.com/basnijholt/lovelace-ios-themes/issues/80#issuecomment-2337950259) above.

Quick way to clean the WebApp cache is to log out and login again, but of course this is a hassle.
